crates/storage/nippy-jar/src/compression/zstd.rs

```diff
@@ -0,0 +1,175 @@
+use crate::{compression::Compression, NippyJarError};
```

Comment:
As I'm reading the compression abstractions I wonder if this should be generalized in any way to also be used in the tables we have on disk, or even more generally across the entire node, given how the ZSTD/LZ4 results via ZFS have been..